File: algorithms/easy/average_value_of_even_numbers_that_are_divisible_by_three.cpp
class Solution {
public:
    int averageValue(vector<int>& nums) {
      vector<int> evens;
      copy_if(nums.begin(), nums.end(), back_inserter(evens),
              [](int x) { return x % 6 == 0; });
      return evens.size() == 0
                 ? 0
                 : accumulate(evens.begin(), evens.end(), 0) / evens.size();
    }
};

File: algorithms/easy/average_value_of_even_numbers_that_are_divisible_by_three.ex
defmodule Solution do
  @spec average_value(nums :: [integer]) :: integer
  def average_value(nums) do
    nums
    |> Enum.filter(&(rem(&1, 6) == 0))
    |> then(fn ns ->
      if ns == [], do: 0, else: ns |> Enum.sum() |> div(length(ns))
    end)
  end
end
